### PR TITLE
Adding field label property for fields

### PR DESCRIPTION
**Pull request by robertjd**
Right now we are using the placeholder value for the label value, which is silly

### CODE REVIEW

File: docs/registration.rst [hunk 1]
```diff
@@ -106,17 +106,34 @@ like this::
...
         }
       }
     }
 
-With the exception of the ``enabled`` property, the rest of the properties will
-be directly applied to the HTML form elements that are created when we render
-the form.
+All field objects have the following properties, which must be defined:
+
+- **Enabled** - determines if the field is shown on the form.
```
Comment (timothyej) [inline docs/registration.rst]: Shouldn't these properties be in lowercase? E.g. `enabled`.

Comment (rdegges) [inline docs/registration.rst]: Yeah, that should be changed.

File: docs/registration.rst [hunk 2]
```diff
@@ -106,17 +106,34 @@ like this::
...
+
+- **Placeholder** - The help text that is shown inside the input field, if the
+  input field is empty (HTML5 property).
+
+- **Required** - Marks the field as a required field.  This uses the HTML5
+  required property, to prompt the user to enter the value.  The post data will
+  also be validated to ensure that the field is supplied, and an error will be
+  returned if the field is empty.
+
+- **Type** - the HTML type of the input, e.g. text, email, or password.
```
Comment (timothyej) [inline docs/registration.rst]: `the` should be `The` to be consistent.

File: docs/registration.rst [hunk 3]
```diff
@@ -106,17 +106,34 @@ like this::
       fields: {
         favoriteColor: {
           enabled: true,
+          label: 'Favorite Color',
           name: 'favoriteColor',
-          placeholder: 'Favorite Color',
+          placeholder: 'e.g. Red, Blue',
```
Comment (typerandom) [inline docs/registration.rst]: Capitalize the `e` in `e.g.`?

**Comment by timothyej** [comment]
Except for my comments this looks good :+1: :)

File: docs/registration.rst (hunk 1 above)
Comment (typerandom) [inline docs/registration.rst]: `determines` should be capitalized.

File: docs/registration.rst (hunk 2 above)
Comment (typerandom) [inline docs/registration.rst]: `the` should be capitalized.

**Comment by typerandom** [comment]
Haha, had this page open but just reviewed it. Didn't see Timothy's comments.

**Pull request merged**